src/ext.rs

```diff
@@ -832,6 +858,31 @@ fn timeout2ms(dur: DWORD) -> Option<u32> {
 }
 
 #[cfg(any(feature = "nightly", feature = "duration"))]
+fn linger2dur(linger_opt: linger) -> Option<Duration> {
+    let dur = Duration::from_secs(linger_opt.l_linger as u64);
+    if dur.as_secs() == 0 {
+        None
+    }
+    else {
+        Some(dur)
+    }
+}
+
+#[cfg(any(feature = "nightly", feature = "duration"))]
+fn dur2linger(dur: Option<Duration>) -> linger {
+    match dur {
+        Some(d) => {
+            let onoff = if d.as_secs() == 0 { 0 } else { 1 };
+            linger {
+                l_onoff: onoff as u16,
```

Comment:
Hmm, I read some [other](http://alas.matf.bg.ac.rs/manuals/lspe/snode=105.html) documentation to make sure I'm not wrong, and it looks like you're right.

I read a lot of threads on stack overflow, and a lot of warnings about this, but it looks like `linger { l_onoff: 1, l_linger:0 }` even has a special meaning for the system. You're right, if we can communitcate it, we should do it. It works on windows.